### ISSUE TITLE
I'm getting "double free or corruption" errors on multiple UTests after make install

### ISSUE THREAD

**Issue by pennachin** [opencog/atomspace#587]
Running the UTests from latest master I get the following failures (snipped `make test` output):

```
      Start 59: ArcanaUTest
*** Error in `/atomspace/build/tests/query/ArcanaUTest': double free or corruption (!prev): 0x00000000028b74a0 ***
59/94 Test #59: ArcanaUTest ......................***Exception: Other  4.22 sec
      Start 61: GetLinkUTest
*** Error in `/atomspace/build/tests/query/GetLinkUTest': double free or corruption (fasttop): 0x0000000002259840 ***
61/94 Test #61: GetLinkUTest .....................***Exception: Other  0.16 sec
      Start 66: EvaluationUTest
*** Error in `/atomspace/build/tests/query/EvaluationUTest': double free or corruption (!prev): 0x0000000000db63d0 ***
66/94 Test #66: EvaluationUTest ..................***Exception: Other  0.22 sec
      Start 79: SudokuUTest
*** Error in `/atomspace/build/tests/query/SudokuUTest': double free or corruption (!prev): 0x0000000001fdc8d0 ***
79/94 Test #79: SudokuUTest ......................***Exception: Other  0.17 sec
      Start 83: ForwardChainerUTest
*** Error in `/atomspace/build/tests/rule-engine/ForwardChainerUTest': double free or corruption (fasttop): 0x000000000269c720 ***
83/94 Test #83: ForwardChainerUTest ..............***Exception: Other  0.52 sec
      Start 86: UREConfigReaderUTest
*** Error in `/atomspace/build/tests/rule-engine/UREConfigReaderUTest': double free or corruption (fasttop): 0x0000000002772170 ***
```

I did a debug build and ran valgrind on ArcanaUTest:

```
valgrind --tool=memcheck --track-origins=yes --keep-stacktraces=alloc-and-free ./tests/query/ArcanaUTest
```

Here's the relevant part of the output:

```
==16314== Invalid free() / delete / delete[] / realloc()
==16314==    at 0x4C2C2BC: operator delete(void*) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==16314==    by 0x52D26B3: __gnu_cxx::new_allocator<opencog::FunctionWrap*>::deallocate(opencog::FunctionWrap**, unsigned long) (new_allocator.h:110)
==16314==    by 0x52D2449: std::_Vector_base<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::_M_deallocate(opencog::FunctionWrap**, unsigned long) (stl_vector.h:174)
==16314==    by 0x52D2320: std::_Vector_base<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::~_Vector_base() (stl_vector.h:160)
==16314==    by 0x52D2CD8: std::vector<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::~vector() (stl_vector.h:416)
==16314==    by 0x621F5E9: __cxa_finalize (cxa_finalize.c:56)
==16314==    by 0x52B5B82: ??? (in /atomspace/build/opencog/atoms/execution/libexecution.so)
==16314==    by 0x4010739: _dl_fini (dl-fini.c:252)
==16314==    by 0x621F258: __run_exit_handlers (exit.c:82)
==16314==    by 0x621F2A4: exit (exit.c:104)
==16314==    by 0x6204ECB: (below main) (libc-start.c:321)
==16314==  Address 0xfe40950 is 0 bytes inside a block of size 128 free'd
==16314==    at 0x4C2C2BC: operator delete(void*) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==16314==    by 0x52D26B3: __gnu_cxx::new_allocator<opencog::FunctionWrap*>::deallocate(opencog::FunctionWrap**, unsigned long) (new_allocator.h:110)
==16314==    by 0x52D2449: std::_Vector_base<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::_M_deallocate(opencog::FunctionWrap**, unsigned long) (stl_vector.h:174)
==16314==    by 0x52D2320: std::_Vector_base<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::~_Vector_base() (stl_vector.h:160)
==16314==    by 0x52D2CD8: std::vector<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::~vector() (stl_vector.h:416)
==16314==    by 0x621F258: __run_exit_handlers (exit.c:82)
==16314==    by 0x621F2A4: exit (exit.c:104)
==16314==    by 0x6204ECB: (below main) (libc-start.c:321)
==16314==  Block was alloc'd at
==16314==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==16314==    by 0x52D296F: __gnu_cxx::new_allocator<opencog::FunctionWrap*>::allocate(unsigned long, void const*) (new_allocator.h:104)
==16314==    by 0x52D27E6: std::_Vector_base<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::_M_allocate(unsigned long) (in /atomspace/build/opencog/atoms/execution/libexecution.so)
==16314==    by 0x52D24C3: void std::vector<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::_M_emplace_back_aux<opencog::FunctionWrap*>(opencog::FunctionWrap*&&) (vector.tcc:404)
==16314==    by 0x52D23BC: void std::vector<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::emplace_back<opencog::FunctionWrap*>(opencog::FunctionWrap*&&) (vector.tcc:101)
==16314==    by 0x52D22A9: std::vector<opencog::FunctionWrap*, std::allocator<opencog::FunctionWrap*> >::push_back(opencog::FunctionWrap*&&) (stl_vector.h:920)
==16314==    by 0x52D2006: opencog::ExecSCM::init() (ExecSCM.cc:100)
==16314==    by 0x552DB84: opencog::ModuleWrap::init_in_module(void*) (SchemeModule.cc:70)
==16314==    by 0x796D1E7: scm_c_with_fluid (in /usr/lib/libguile-2.0.so.22.7.0)
==16314==    by 0x798BF17: scm_c_define_module (in /usr/lib/libguile-2.0.so.22.7.0)
==16314==    by 0x552DB45: opencog::ModuleWrap::init_in_guile(void*) (SchemeModule.cc:59)
==16314==    by 0x795B2C9: ??? (in /usr/lib/libguile-2.0.so.22.7.0)
```

**Comment by linas** [comment]
#588 should fix this

**Comment by pennachin** [comment]
Fixed in #589, thanks.
